static/app/utils/replays/hooks/useReplayViewedByData.tsx

```diff
@@ -0,0 +1,33 @@
+import type {User} from 'sentry/types';
+import {
+  type ApiQueryKey,
+  useApiQuery,
+  type UseApiQueryOptions,
+} from 'sentry/utils/queryClient';
+
+interface Props {
+  orgSlug: string;
+  projectSlug: undefined | string;
+  replayId: undefined | string;
```

Comment:
Why is `undefined` allowed for replayId and projectSlug? Why not for orgSlug?

Reply:
@aliu3ntry it comes from the `replayMetaData.tsx` file that calls this hook. `replayRecord` might not exist, but that's where we get the `projectSlug` and `replayId` from

`projectSlug: replayRecord?.project_id,`

Reply:
Got it, what happens if undefined is passed to the apiQuery? Will we use a network call calling the nonexistent endpoint `"projects/undefined/...."`?

Reply:
fixed the `projects/undefined/....` problem. good catch @andrewshie-sentry!

Reply:
thanks! wrong andrew :(